packages/eas-cli/src/metadata/config.ts

```diff
@@ -12,6 +18,83 @@ export interface MetadataConfig {
   apple?: AppleMetadata;
 }
 
+/**
+ * Resolve the dynamic config from the user.
+ * It supports methods, async methods, or objects (json).
+ */
+async function resolveDynamicConfigAsync(configFile: string): Promise<unknown> {
+  const userConfigContent = await import(configFile);
+  let userConfig = userConfigContent.default || userConfigContent;
+
+  if (typeof userConfig === 'function') {
+    userConfig = await userConfig();
+  }
```

Comment:
Maybe sth like:
```suggestion
  let maybeUserConfigFunction = userConfigContent.default ?? userConfigContent;

  let userConfig;
  if (typeof maybeUserConfigFunction === 'function') {
    userConfig = await maybeUserConfigFunction();
  }
```

Reply:
Seems good to me. I think in other places I've used something like `xOrY`, so I'll change it to:

```
const userConfigOrFunction = ...
let userConfig
```

Does that seem reasonable?

Reply:
Yes, makes sense.